src/pal/src/include/pal/threadsusp.hpp

```diff
@@ -73,6 +73,8 @@ Module Name:
 #include <sys/sem.h>
 #include <sys/types.h>
 
+#elif __ANDROID__
+#include <semaphore.h>
```

Comment:
Going by the NDK's `semaphore.h` file, seems like it should be  `#ifdef _SEMAPHORE_H`. :)

Regarding the cmake check, I *think* (haven't tested myself) the following patch would work in `src/pal/src/configure.cmake`:

```cmake
include(CheckIncludeFileCXX)
check_include_file_cxx(semaphore.h HAVE_SEMAPHORE_H) # C++ instead of C
```